Proposed/asc058.md

```diff
@@ -0,0 +1,40 @@
+#### **Title**
+Add `attach_sourcing_info` as a datatype for the `informatics` instruction attribute
+
+#### **Authorship**
+Al Colmenar <al.colmenar@strateos.com>
+
+#### **Motivation**
+Currently, certain instructions operate on resources that don't contain the information needed to further drill down into the sourcing information associated to a particular container. This additional sourcing information is required when a customer wants to provide the vendor and supplier of the resource to be used in a run. An example of an instruction where this would be useful is the `provision` Autoprotocol instruction which is meant to describe a task of provisioning a container for a given resource.
```

Comment:
I'm not sure I am entirely clear on the motivation here.

> instructions operate on resources that don't contain the information needed to further drill down into the sourcing information associated to a particular container. This additional sourcing information is required when a customer wants to provide the vendor and supplier of the resource to be used in a run

Are you referring to resource IDs? If we are worried about a resource ID not pointing to the correct vendor/resource, we can make a new resource that is more specific.

Reply:
The issue with resource model is that the supplier and vendor information actually isn't part of it. In the future, I think that we'd have to come together to decide how to address the issue of where the supplier/vendor information should exist for a resource (currently it is in kit). 

Also, I'm not entirely sure the provision instruction should be operating on a Strateos defined resource id and if we move away from this to a more generic naming of the resource, e.g. having the resource name as the provision parameter, this change would help facilitate a user's desire to determine the sourcing information of a generic resource name

Reply:
If there are issues with the resource model, why don't we update it?

If we specify vendor and supplier in autoprotocol, but then rely on WEB to find a resource with those attributes, won't that require an update to the Resource model as well?

Reply:
> I'm not entirely sure the provision instruction should be operating on a Strateos defined resource id 
There is an interesting discussion to be had here. However, if we pass a string instead of a UUID, the onus of entity lookup falls on the consumer of the autoprotocol. At least now, whoever consumes autoprotocol can do their own record keeping. If they want to pass in a resource string `"water"` and that means something to them, great! But if we're passing this string, and a number of filters in a dictionary, then we're prescribing how autoprotocol should be consumed.

> if we move away from this to a more generic naming of the resource, e.g. having the resource name as the provision parameter, this change would help facilitate a user's desire to determine the sourcing information of a **generic resource name**
I still don't understand what you mean by "generic resource name." All resources are unique. They all have a unique ID.